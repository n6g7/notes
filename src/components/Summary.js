import React, { PureComponent } from 'react'
import { Link } from 'react-router'

import notebooks from '../../notebooks'

class Summary extends PureComponent {
  render () {
    return <div>
      <h1>summary</h1>
      <ul>
        { Object.keys(notebooks).map(notebook =>
          <li key={`summary-${notebook}`}>
            <Link to={`/nb/${notebook}`}>{notebook}</Link>
          </li>
        ) }
      </ul>
    </div>
  }
}

export default Summary
