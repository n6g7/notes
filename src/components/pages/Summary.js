import React, { PureComponent } from "react"
import { Link } from "react-router-dom"

import notebooks from "@notebooks"
import { Page } from "@templates"

class Summary extends PureComponent {
  render() {
    return (
      <Page>
        <h1>summary</h1>
        <ul>
          {Object.keys(notebooks).map((notebook) => (
            <li key={`summary-${notebook}`}>
              <Link to={`/nb/${notebook}`}>{notebook}</Link>
            </li>
          ))}
        </ul>
      </Page>
    )
  }
}

export default Summary
