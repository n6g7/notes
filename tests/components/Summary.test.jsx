import React from 'react'
import Summary from '../../src/components/Summary.jsx'
import renderer from 'react-test-renderer'

describe('Summary', () => {
  it('renders a well formed component', () => {
    const component = renderer.create(<Summary />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
