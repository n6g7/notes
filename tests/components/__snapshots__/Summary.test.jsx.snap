exports[`Summary renders a well formed component 1`] = `
<div>
  <h1>
    summary
  </h1>
  <ul>
    <li>
      <a
        onClick={[Function]}
        style={Object {}}>
        accelerating-returns
      </a>
    </li>
    <li>
      <a
        onClick={[Function]}
        style={Object {}}>
        bertrand
      </a>
    </li>
    <li>
      <a
        onClick={[Function]}
        style={Object {}}>
        secretary
      </a>
    </li>
    <li>
      <a
        onClick={[Function]}
        style={Object {}}>
        simulation
      </a>
    </li>
  </ul>
</div>
`;
